File: scripts/player.lua
player = {}

player.x = 10
player.y = 200
player.speed = 200
player.direction = 1 -- 1 for right, -1 for left

player.scale_x = 2
player.scale_y = 2

player.image_idle = love.graphics.newImage('data/images/player/idle.png')
player.image_run = love.graphics.newImage('data/images/player/run.png')

function player.update(dt)
    local dir = 0

    if love.keyboard.isDown('d') then -- use d to move right
        dir = dir + 1
    end

    if love.keyboard.isDown('a') then -- use a to move left
        dir = dir - 1
    end

    if dir ~= 0 then
        player.direction = dir
    end

    player.x = player.x + dir * player.speed * dt
end

function player.draw()
    local img = player.image_idle
    local ox = img:getWidth() / 2
    local oy = img:getHeight() / 2

    love.graphics.draw(img, player.x, player.y, 0, -player.scale_x * player.direction, player.scale_y, ox, oy) -- -player.scale_x to account for the image facing left 
end

File: scripts/player.py
import pygame

class Player:
    def __init__(self):
        self.x = 10
        self.y = 200
        self.speed = 200
        self.direction = 1  # 1 for right, -1 for left
        self.scale_x = 2
        self.scale_y = 2
        
        # Create simple colored rectangles for now (placeholder for images)
        self.width = 32
        self.height = 32
        self.color = (255, 255, 255)  # White color for player
        
    def update(self, dt):
        keys = pygame.key.get_pressed()
        dir_x = 0
        
        if keys[pygame.K_d]:  # Move right
            dir_x += 1
        if keys[pygame.K_a]:  # Move left
            dir_x -= 1
            
        if dir_x != 0:
            self.direction = dir_x
            
        self.x += dir_x * self.speed * dt
        
    def draw(self, screen):
        # Draw player as a rectangle (placeholder for sprite)
        draw_width = self.width * self.scale_x
        draw_height = self.height * self.scale_y
        
        # Flip direction if facing left
        if self.direction == -1:
            pygame.draw.rect(screen, self.color, 
                           (self.x - draw_width//2, self.y - draw_height//2, draw_width, draw_height))
        else:
            pygame.draw.rect(screen, self.color, 
                           (self.x - draw_width//2, self.y - draw_height//2, draw_width, draw_height))